### PR TITLE
Github issue 790: Validate QueueFamilyIndices across CB creation and submission

### PR DESCRIPTION
**Pull request by mark-lunarg**

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -4682,6 +4682,42 @@ static bool validateCommandBufferState(layer_data *dev_data, GLOBAL_CB_NODE *pCB
...
+// Validate that queueFamilyIndices of primary and any secondary command buffers match this queue
+static bool validateQueueFamilyIndices(layer_data *dev_data, GLOBAL_CB_NODE *pCB, VkQueue queue) {
+    bool skip_call = false;
+    auto pPool = getCommandPoolNode(dev_data, pCB->createInfo.commandPool);
+    auto queue_node = getQueueNode(dev_data, queue);
+
+    if (pPool && queue_node && (pPool->queueFamilyIndex != queue_node->queueFamilyIndex)) {
+        skip_call |= log_msg(dev_data->report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_COMMAND_BUFFER_EXT,
+            reinterpret_cast<uint64_t>(pCB->commandBuffer), __LINE__, DRAWSTATE_INVALID_QUEUE_FAMILY, "DS",
+            "vkQueueSubmit: Primary ommand buffer 0x%" PRIxLEAST64
```
Comment (chrisforbes) [inline layers/core_validation.cpp]: s/ommand/command/

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -4682,6 +4682,42 @@ static bool validateCommandBufferState(layer_data *dev_data, GLOBAL_CB_NODE *pCB
...
+            " created in queue family %d is being submitted on queue 0x%" PRIxLEAST64 " from queue family %d.",
+            reinterpret_cast<uint64_t>(pCB->commandBuffer), pPool->queueFamilyIndex,
+            reinterpret_cast<uint64_t>(queue), queue_node->queueFamilyIndex);
+    }
+
+    if (!pCB->secondaryCommandBuffers.empty()) {
```
Comment (chrisforbes) [inline layers/core_validation.cpp]: Is this the right time to do it, or can/should this be done at `CmdExecuteCommands` time instead?

Comment (chrisforbes) [inline layers/core_validation.cpp]: Extra check for empty around here seems spurious? Not a big deal though

File: layers/core_validation.cpp [hunk 3]
```diff
@@ -4682,6 +4682,42 @@ static bool validateCommandBufferState(layer_data *dev_data, GLOBAL_CB_NODE *pCB
...
+        for (auto secondaryCmdBuffer : pCB->secondaryCommandBuffers) {
+            GLOBAL_CB_NODE *pSubCB = getCBNode(dev_data, secondaryCmdBuffer);
+            if (pSubCB) {
+                pPool = getCommandPoolNode(dev_data, pSubCB->createInfo.commandPool);
```
Comment (chrisforbes) [inline layers/core_validation.cpp]: Seems unfortunate that we have to chase this through the pool, but we do lots of crazy stuff at submit time...

**Comment by chrisforbes** [comment]
:+1:

File: layers/core_validation.cpp (hunk 2 above)
Comment (mark-lunarg) [inline layers/core_validation.cpp]: There is valid-usage language in CmdExecuteCommands for this, so I'll move the secondary-CB checking into validateSecondaryCommandBufferState and only check the primary at submit time -- great catch!  I'll make sure there's no check-for-empty in the new stuff.

File: layers/core_validation.cpp (hunk 3 above)
Comment (mark-lunarg) [inline layers/core_validation.cpp]: Yes, maybe in the future...

**Comment by mark-lunarg** [comment]
Pushed to Master.

**Pull request closed without merging**